server/src/main/java/org/candlepin/auth/KeycloakAdapterConfiguration.java

```diff
@@ -42,18 +41,23 @@
     private KeycloakDeployment keycloakDeployment;
 
     @Inject
-    public KeycloakAdapterConfiguration(Configuration configuration) {
-        if (configuration.getBoolean(ConfigProperties.KEYCLOAK_AUTHENTICATION)) {
-            try {
-                adapterConfig = KeycloakDeploymentBuilder.loadAdapterConfig(new
-                        FileInputStream(configuration.getString(ConfigProperties.KEYCLOAK_FILEPATH)));
-                keycloakDeployment = KeycloakDeploymentBuilder.build(adapterConfig);
+    public KeycloakAdapterConfiguration(Configuration configuration) throws Exception {
+        if (configuration.getBoolean(ConfigProperties.KEYCLOAK_AUTHENTICATION, false)) {
+            String configFile = configuration.getString(ConfigProperties.KEYCLOAK_FILEPATH, null);
+
+            try (InputStream cfgStream = new FileInputStream(configFile)) {
+                adapterConfig = KeycloakDeploymentBuilder.loadAdapterConfig(cfgStream);
+                this.keycloakDeployment = KeycloakDeploymentBuilder.build(adapterConfig);
             }
-            catch (FileNotFoundException e) {
-                log.warn("Keycloak.json file not found", e);
+            catch (IOException e) {
+                log.error("Unable to read keycloak configuration file: {}", configFile);
+                throw e;
             }
-            catch (RuntimeException e) {
-                log.warn("Unable to read keycloak.json", e);
+            catch (Exception e) {
```

Comment:
One thing that needs to be retained that I didn't cover in my example was the RuntimeException handling, since that's what Keycloak uses for bad configuration exceptions.